src/3d/shaders/diffuseSpecular.vert

```diff
@@ -0,0 +1,28 @@
+#version 330
+
+in vec3 vertexPosition;
+in vec3 vertexNormal;
+in vec2 vertexTexCoord;
+
+out vec3 worldPosition;
+out vec3 worldNormal;
+out vec2 texCoord;
+
+uniform mat4 modelMatrix;
+uniform mat3 modelNormalMatrix;
+uniform mat4 modelViewProjection;
+
+uniform float texCoordScale;
+
+void main(void)
+{
+    // Pass through scaled texture coordinates
+    texCoord = vertexTexCoord * texCoordScale;
+
+    // Transform position and normal to world space
+    worldPosition = vec3(modelMatrix * vec4(vertexPosition, 1.0));
+    worldNormal = normalize(modelNormalMatrix * vertexNormal);
+
+    // Calculate vertex position in clip coordinates
+    gl_Position = modelViewProjection * vec4(vertexPosition, 1.0);
+}
```

Comment:
Where does this shader come from - is it a copy of src/extras/shaders/gl3/default.vert from qt3d? If so, It would be good to note that in a comment in the header (or at least to have a note it is based on that file with what kind of modifications)

Reply:
I have added a comment at the beginning of `diffuseSpecular.vert`